compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -95,58 +95,79 @@ object DesugarEnums {
   /**  The following lists of definitions for an enum type E and known value cases e_0, ..., e_n:
    *
    *   private val $values = [e_0, ..., e_n : E]
-   *   @annotation.threadUnsafe private lazy val $valuesReverse =
-   *     scala.runtime.ScalaRuntime.wrapRefArray($values).map((x_0: E) => (x_0.enumLabel, x_0)).toMap
    *   def values = $values.clone
    *   def valueOf($name: String) =
-   *     try $valuesReverse($name) catch
-   *       {
-   *         case ex$: NoSuchElementException =>
-   *           throw new IllegalArgumentException("enum case not found: " + $name)
-   *       }
+   *     val nameHash = if $name == null then 0 else $name.hashCode
```

Comment:
I would just do a pattern match on strings and worry about optimizing those later.

Reply:
will do that